task/runner.go

```diff
@@ -1,70 +1,58 @@
 package task
 
 import (
-	"encoding/json"
-	"fmt"
-	"github.com/google/uuid"
-	"github.com/turbot/steampipe/constants"
-	"github.com/turbot/steampipe/db"
-	"io/ioutil"
-	"os"
-	"path/filepath"
+	"sync"
 	"time"
+
+	"github.com/turbot/steampipe/db"
+	"github.com/turbot/steampipe/statefile"
 )
 
 const minimumMinutesBetweenChecks = 1440 // 1 day
-const updateStateFileName = "update-check.json"
 
 type Runner struct {
-	currentState state
+	currentState *statefile.State
 	shouldRun    bool
 }
 
 func NewRunner() *Runner {
 	r := new(Runner)
-	r.loadState()
+	r.currentState, _ = statefile.LoadState()
 	r.shouldRun = r.getShouldRun()
 	return r
 }
 
 func (r *Runner) Run() {
 	if r.shouldRun {
+		waitGroup := sync.WaitGroup{}
+		waitGroup.Add(3)
```

Comment:
I'd rather have a separate `waitGroup.Add(1)` begore each go routine - that way if we add another we will hopefully remember to inc the wg